### PR TITLE
Ticket system

### PR DESCRIPTION
**Pull request by spark6796**

### CODE REVIEW

**Comment by spark6796** [comment]
ticket system

File: cogs/ticket.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
+
+import disnake
+from disnake.ext import commands
+from datetime import datetime
+from core.utils import TicketCog
```
Comment (externref) [inline cogs/ticket.py:5]: there's no TicketCog in utils.py
you meant HelperCog ig?

File: cogs/ticket.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+from disnake.ext import commands
+from datetime import datetime
+from core.utils import TicketCog
+from core.constants import RoleID , ChannelID
+MODERATOR_ROLE_ID = RoleID.MODERATOR
+TICKET_CHANNEL_ID = ChannelID.TICKET_CHANNEL_ID
+class ticket(TicketCog):
+    """tickets"""
+    def __init__(self , bot: commands.Bot):
+        self.bot = bot
```
Comment (externref) [inline cogs/ticket.py]: not required, its done internally

File: cogs/ticket.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    # Update ticket name when user changes name
+    @commands.Cog.listener()
+    async def on_member_update(self , before : disnake.Member , after : disnake.Member):
+        username_before : str = before.name+before.discriminator
+        username_after : str = after.name+after.discriminator
```
Comment (externref) [inline cogs/ticket.py]:
```py
str(user) 
``` 
gives you the name#discrim format

File: cogs/ticket.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+                )
+            await self.thread.remove_user(self.author)
+            await self.thread.edit(name=f"Closed-{self.author.name+self.author.discriminator}")
+    async def on_timeout(self):
+        await self.thread.delete()
+
+
+
+def setup(bot: commands.Bot):
+    bot.add_cog(ticket(bot))
```
Comment (externref) [inline cogs/ticket.py]: setup is not required here

**Review by externref — CHANGES_REQUESTED** [review]

File: cogs/ticket.py (hunk 1 above)
Comment (spark6796) [reply cogs/ticket.py:5]: um yeah i didnt setup that

**Pull request closed without merging**